src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -39,9 +40,48 @@ public EventsourcedState(string name, bool isRecoveryRunning, StateReceive state
     public abstract partial class Eventsourced
     {
         /// <summary>
-        /// Processes a loaded snapshot, if any. A loaded snapshot is offered with a <see cref="SnapshotOffer"/> 
-        /// message to the actor's <see cref="ReceiveRecover"/>. Then initiates a message replay, either starting 
-        /// from the loaded snapshot or from scratch, and switches to <see cref="ReplayStarted"/> state. 
+        /// TBD
+        /// </summary>
+        /// <param name="action">TBD</param>
+        protected void RunTask(Func<Task> action)
+        {
+            Func<Task> wrap = () =>
+            {
+                Task t = action();
+                if (!t.IsCompleted)
+                {
+                    var tcs = new TaskCompletionSource<object>();
+                    t.ContinueWith(r =>
```

Comment:
Before I didn't need it, but now i need to react also on failure.
Regarding Context, it works as usual. Context, Sender,.. are accessible inside the command handler. Same for Persist callback except of the Sender, but that's by design